OutOfSchool/OutOfSchool.WebApi/Controllers/V1/BaseMementoController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using OutOfSchool.BusinessLogic.Common;
+using OutOfSchool.BusinessLogic.Services.Memento.Interfaces;
+
+namespace OutOfSchool.WebApi.Controllers.V1;
+
+/// <summary>Base Controller with operations for storing data in cache.</summary>
+/// <typeparam name="T">T is the entity type that should be stored in the cache.</typeparam>
+public abstract class BaseMementoController<T> : ControllerBase
+{
+    private readonly IMementoService<T> mementoService;
+    private readonly IStorage storage;
+
+    /// <summary>Initializes a new instance of the <see cref="BaseMementoController{T}" /> class.</summary>
+    /// <param name="mementoService">The memento service.</param>
+    /// <param name="storage">The storage.</param>
+    /// <exception cref="System.ArgumentNullException">crudCacheService
+    /// or
+    /// mementoService
+    /// or
+    /// storage.</exception>
+    protected BaseMementoController(
+        IMementoService<T> mementoService,
+        IStorage storage)
+    {
+        this.mementoService = mementoService ?? throw new ArgumentNullException(nameof(mementoService));
+        this.storage = storage ?? throw new ArgumentNullException(nameof(storage));
```

Comment:
(Logic) Controllers should usually avoid containing logic. In this controller you have 2 services and introduce logic of calling both of them in specific order. As for me it looks like they are highly coupled, that shouldn't be (remember, _high cohesion, low coupling_). 
Probably something is wrong on design level and these 2 services should be the one (or at least hidden by the facade of one additional service)?

Reply:
It's done. 
Storage has been removed from code.
All CRUD logic of cache is relocated to ICrudCacheService.